cpp/documentation/README.md

```diff
@@ -0,0 +1,460 @@
+# C++ Project Documentation
+
+This document provides instructions and hints to document C++ projects.
+
+## How to document code?
+
+### C++ code
+
+Public APIs may be documented with Doxygen. Documentation of private code is recommended.
+
+#### Documentation style
+
+Doxygen supports [multiple syntax](http://www.doxygen.nl/manual/docblocks.html)
+to document the code. There is no preferred syntax but Doxygen comments of a particular
+project may use the same one.
+
+Nevertheless, it is suggested to turn the
+[JAVADOC_AUTOBRIEF](http://www.doxygen.nl/manual/config.html#cfg_javadoc_autobrief)
+option to `YES` to shrink comments size:
+
+```c++
+/// A brief function description.
+/// A more elaborated function description.
+void a_function();
+```
+
+instead of:
+```c++
+/// \brief A brief function description.
+///
+/// A more elaborated function description.
+void a_function();
+
```

Comment:
@tristan0x : thinking more about this, consider below example : 

![Screenshot 2019-04-15 at 20 28 47](https://user-images.githubusercontent.com/666852/56156282-3722dc80-5fbd-11e9-8bbb-70b3cd15f1c9.png)

Brief description is often can not be read together with the elaborated paragraph (if we write one-after-another, it seems like continuous statements). In that case, newline between two seems useful (semantically and visually). Obviously we could avoid `\brief` itself.

What do you think?

Reply:
I am not against adding a newline to separate the 2 parts.